financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/dcr/validators/TokenEndpointAuthSigningAlgValidator.java

```diff
@@ -0,0 +1,86 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.ServiceProviderProperty;
+import org.wso2.carbon.identity.oauth.dcr.bean.ApplicationRegistrationRequest;
+import org.wso2.carbon.identity.oauth.dcr.bean.ApplicationUpdateRequest;
+import org.wso2.financial.services.accelerator.common.exception.FinancialServicesException;
+
+import java.util.Map;
+
+/**
+ * Validator class for validating he token endpoint authentication signing algorithm is available and not empty when
+ * token endpoint authentication method is private ket jwt.
+ */
+public class TokenEndpointAuthSigningAlgValidator implements DynamicClientRegistrationValidator {
+
+    private static final Log log = LogFactory.getLog(TokenEndpointAuthSigningAlgValidator.class);
+    private static final String AUTH_METHOD_PRIVATE_KEY_JWT = "private_key_jwt";
+
+    @Override
+    public void validatePost(ApplicationRegistrationRequest applicationRegistrationRequest,
+                             Map<String, Object> ssaParams) throws FinancialServicesException {
+
+        String tokenEndpointAuthMethod = applicationRegistrationRequest.getTokenEndpointAuthMethod();
+        String tokenEndpointAuthSigningAlg = applicationRegistrationRequest.getTokenEndpointAuthSignatureAlgorithm();
+        validateTokenEPAuthSigningAlg(tokenEndpointAuthMethod, tokenEndpointAuthSigningAlg);
+    }
+
+    @Override
+    public void validateGet(Map<String, String> ssaParams) throws FinancialServicesException {
+
+    }
+
+    @Override
+    public void validateUpdate(ApplicationUpdateRequest applicationUpdateRequest, Map<String, Object> ssaParams,
+                               ServiceProviderProperty[] serviceProviderProperties) throws FinancialServicesException {
+
+
+        String tokenEndpointAuthMethod = applicationUpdateRequest.getTokenEndpointAuthMethod();
+        String tokenEndpointAuthSigningAlg = applicationUpdateRequest.getTokenEndpointAuthSignatureAlgorithm();
+        validateTokenEPAuthSigningAlg(tokenEndpointAuthMethod, tokenEndpointAuthSigningAlg);
+    }
+
+    /**
+     * Validate whether the token endpoint authentication signing algorithm is available and not empty when
+     * token endpoint authentication method is private ket jwt.
+     *
+     * @param tokenEndpointAuthMethod        Token endpoint authentication method
+     * @param tokenEndpointAuthSigningAlg    Token endpoint authentication signing algorithm
+     * @throws FinancialServicesException When the issuer is not equal to the software id
+     */
+    private static void validateTokenEPAuthSigningAlg(String tokenEndpointAuthMethod,
+                                          String tokenEndpointAuthSigningAlg) throws FinancialServicesException {
+
+        // token_endpoint_auth_signing_alg must be specified if token_endpoint_auth_method is private_key_jwt.
+        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod) &&
```

Comment:
Isn't this covered by `StringUtils.equals(tokenEndpointAuthMethod, AUTH_METHOD_PRIVATE_KEY_JWT)` line?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/327/commits/9bdd95d9a4517faed8c734812dfcd9065d731ce0